service/resource/kmskeyv2/delete.go

```diff
@@ -10,24 +10,27 @@ import (
 )
 
 func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
-	deleteAliasInput, err := toDeleteAliasInput(deleteChange)
+	deleteInput, err := toKMSKeyState(deleteChange)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	keyAlias := deleteAliasInput.AliasName
-	if keyAlias != nil && *keyAlias != "" {
-		if _, err := r.awsClients.KMS.DeleteAlias(&deleteAliasInput); err != nil {
-			return microerror.Mask(err)
-		}
-
+	if deleteInput.KeyAlias != "" {
+		// Get the KMS key ID using the key alias.
```

Comment:
Now we get the key Arn using the alias before deleting it. So we can be sure we have the right value.